x/ibc/23-commitment/keeper/keeper.go

```diff
@@ -0,0 +1,82 @@
+package keeper
+
+import (
+	"bytes"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/ibc/23-commitment/exported"
+	"github.com/cosmos/cosmos-sdk/x/ibc/23-commitment/types"
+)
+
+// Keeper defines the IBC commitment keeper (i.e the vector commitment manager).
+// A vector commitment manager has the ability to add or remove items from the
+// commitment state as defined in https://github.com/cosmos/ics/tree/master/spec/ics-023-vector-commitments#definitions.
+type Keeper struct {
+	prefix              exported.PrefixI
+	verifiedMemberships map[string][]byte // lookup map for returning already verified membership proofs
```

Comment:
Should these maps be cleared after each new block?

Otherwise, elements could be added/removed in the next block but the cache will be returning wrong values

Reply:
I think I'll remove the Keeper since it's actually not used and refactor the batch verification to not rely on already verified memberships